osu.Game/Users/Drawables/ClickableAvatar.cs

```diff
@@ -4,90 +4,54 @@
 
 namespace osu.Game.Users.Drawables
 {
-    public partial class ClickableAvatar : Container
+    public partial class ClickableAvatar : OsuClickableContainer
     {
         private const string default_tooltip_text = "view profile";
 
-        /// <summary>
-        /// Whether to open the user's profile when clicked.
-        /// </summary>
-        public bool OpenOnClick
-        {
-            set => clickableArea.Enabled.Value = clickableArea.Action != null && value;
-        }
+        public override LocalisableString TooltipText { get; set; } = default_tooltip_text;
 
         /// <summary>
         /// By default, the tooltip will show "view profile" as avatars are usually displayed next to a username.
         /// Setting this to <c>true</c> exposes the username via tooltip for special cases where this is not true.
         /// </summary>
         public bool ShowUsernameTooltip
         {
-            set => clickableArea.TooltipText = value ? (user?.Username ?? string.Empty) : default_tooltip_text;
+            set => TooltipText = value ? (user?.Username ?? string.Empty) : default_tooltip_text;
         }
 
         private readonly APIUser user;
 
         [Resolved(CanBeNull = true)]
         private OsuGame game { get; set; }
 
-        private readonly ClickableArea clickableArea;
-
         /// <summary>
         /// A clickable avatar for the specified user, with UI sounds included.
-        /// If <see cref="OpenOnClick"/> is <c>true</c>, clicking will open the user's profile.
         /// </summary>
         /// <param name="user">The user. A null value will get a placeholder avatar.</param>
         public ClickableAvatar(APIUser user = null)
         {
             this.user = user;
 
-            Add(clickableArea = new ClickableArea
-            {
-                RelativeSizeAxes = Axes.Both,
-            });
-
             if (user?.Id != APIUser.SYSTEM_USER_ID)
-                clickableArea.Action = openProfile;
+                Action = openProfile;
         }
 
         [BackgroundDependencyLoader]
         private void load()
         {
-            LoadComponentAsync(new DrawableAvatar(user), clickableArea.Add);
+            LoadComponentAsync(new DrawableAvatar(user), Add);
         }
 
         private void openProfile()
         {
             if (user?.Id > 1 || !string.IsNullOrEmpty(user?.Username))
                 game?.ShowUser(user);
         }
-
-        private partial class ClickableArea : OsuClickableContainer
-        {
-            private LocalisableString tooltip = default_tooltip_text;
-
-            public override LocalisableString TooltipText
-            {
-                get => Enabled.Value ? tooltip : default;
-                set => tooltip = value;
-            }
-
-            protected override bool OnClick(ClickEvent e)
-            {
-                if (!Enabled.Value)
-                    return false;
-
-                return base.OnClick(e);
-            }
```

Comment:
These removals are causing behavioural changes for guest users. Most notably on the leaderboard.

Current master:

https://user-images.githubusercontent.com/20418176/217953341-6ae13133-2983-4917-a1b7-2cd939afe146.mp4

Guest user has: no tooltip, no hover sounds, lets through clicks to the background so score is shown.

This PR:

https://user-images.githubusercontent.com/20418176/217953687-3d7cd840-394b-42af-84b9-46a9dddb0645.mp4

Guest user has "show profile" tooltip, has no hover sounds, clicking does nothing.

Feels like a UX regression to me.